lib/src/rendering/pretty_qr_render_view.dart

```diff
@@ -0,0 +1,183 @@
+  @protected
+  DecorationImagePainter? _decorationImagePainter;
+
+  /// Creates a pretty qr view.
+  PrettyQrRenderView({
+    required QrImage qrImage,
+    required PrettyQrDecoration decoration,
+    final RenderBox? child,
+    final ImageConfiguration configuration = ImageConfiguration.empty,
+  })  : _qrImage = qrImage,
+        _decoration = decoration,
+        _configuration = configuration,
+        super(child);
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.qrImage}
+  QrImage get qrImage {
+    return _qrImage;
+  }
+
+  /// Sets the qr image.
+  set qrImage(QrImage value) {
+    if (_qrImage == value) return;
+
+    _qrImage = value;
+    markNeedsPaint();
+  }
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.decoration}
+  PrettyQrDecoration get decoration {
+    return _decoration;
+  }
+
+  /// Sets the current decoration.
+  set decoration(PrettyQrDecoration value) {
+    if (_decoration == value) return;
+
+    if (_decoration.image?.image != value.image?.image) {
+      _decorationImagePainter?.dispose();
+      _decorationImagePainter = null;
+    }
+
+    _decoration = value;
+    markNeedsPaint();
+  }
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.configuration}
+  ImageConfiguration get configuration {
+    return _configuration;
+  }
+
+  /// Sets the image configuration.
+  set configuration(ImageConfiguration value) {
+    if (value == _configuration) return;
+
+    _configuration = value;
+    markNeedsPaint();
+  }
+
+  @override
+  bool hitTestSelf(Offset position) {
+    return (Offset.zero & size).contains(position);
+  }
+
+  @override
+  Size computeSizeForNoChild(BoxConstraints constraints) {
+    return Size.square(min(constraints.maxWidth, constraints.maxHeight));
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final canvas = context.canvas;
+    canvas.save();
+
+    if (offset != Offset.zero) {
+      canvas.translate(offset.dx, offset.dy);
+    }
+
+    final paintingContext = PrettyQrPaintingContext(
+      canvas: canvas,
+      bounds: Offset.zero & size,
+      matrix: PrettyQrMatrix.fromQrImage(qrImage),
+      textDirection: configuration.textDirection,
+    );
+
+    final image = decoration.image;
+    if (image != null) {
+      final imageRect = Rect.fromCenter(
+        center: size.center(Offset.zero),
+        width: size.width * image.scale,
+        height: size.height * image.scale,
+      );
+
+      final imagePadding = image.padding.resolve(
+        configuration.textDirection,
+      );
+
+      // clear space for the embedded image
+      if (image.position == PrettyQrDecorationImagePosition.embedded) {
+        final imageClippedRect = imagePadding.inflateRect(imageRect);
+        for (final module in paintingContext.matrix) {
+          final moduleRect = module.resolve(paintingContext);
+
+          if (imageClippedRect.overlaps(moduleRect)) {
+            paintingContext.matrix.removeDarkAt(module.x, module.y);
```

Comment:
removeDarkAt -> setWhite?

Reply:
I don't quite agree that naming it `setWhite` is a more appropriate variant. What about `removeDarkModuleAt`?